### PR TITLE
Integrate YPP authorization with Google

### LINKED ISSUE #3155: Integrate YPP authorization with Google
**Scope:**

- [ ] Integrate Google authentication with YPP authorization flow once the spike (#3152 is done)
- [ ] Handle Google authentication failure (as seen https://www.figma.com/file/oQqFqdAiPu16eeE2aA5AD5/YouTube-Partner-Program?node-id=1818%3A90535)

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #3155

### CODE REVIEW

**Review by rafalpawlow — COMMENTED** [review]
@kdembler It looks like something is wrong with google url

```url
https://accounts.google.com/o/oauth2/v2/auth?client_id=812154685698-u8egldgkcuuuftmba181ffp3nnuhauat.apps.googleusercontent.com&response_type=code&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fyoutube.force-ssl&include_granted_scopes=true&prompt=consent&redirect_uri=https%3A%2F%2Fatlas-o7r2640hl-joystream.vercel.app%2Fypp&state=code%3Duw9g2kyv2%26channelId%3D2
```

Zrzut ekranu 2022-09-28 o 19 14 47

**Comment by rafalpawlow** [comment]
> @kdembler It looks like something is wrong with google url
> 
> ```ini
> https://accounts.google.com/o/oauth2/v2/auth?client_id=812154685698-u8egldgkcuuuftmba181ffp3nnuhauat.apps.googleusercontent.com&response_type=code&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fyoutube.force-ssl&include_granted_scopes=true&prompt=consent&redirect_uri=https%3A%2F%2Fatlas-o7r2640hl-joystream.vercel.app%2Fypp&state=code%3Duw9g2kyv2%26channelId%3D2
> ```
> 
> Zrzut ekranu 2022-09-28 o 19 14 47

Oh, I forgot we need to add it to google console 🤦

File: packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.hooks.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      SentryLogger.error('Google Auth failed', 'YppAuthorizationModal', error)
+
+      // TODO: use proper copy
+      openConfirmationModal({
+        title: 'Authorization failed',
+        description: <>Google Auth failed</>,
+        type: 'destructive',
+        primaryButton: {
+          text: 'Try again',
+          onClick: () => {
```
Comment (rafalpawlow) [inline packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.hooks.tsx:76]: Shouldn't we start authorization flow from the beginning? In fact it does the same as cancel button

File: packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationSteps/YppAuthorizationRequirementsStep/YppAuthorizationRequirementsStep.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    if (!isChannelValid) {
+      displaySnackbar({
+        title: `Your Atlas channel doesn't meet conditions`,
+        description:
+          'Your Atlas channel must have a custom avatar, cover image, and description set in order to be enrolled in the program.',
+        iconType: 'error',
+        actionText: 'Edit channel',
+        actionIcon: <SvgActionNewTab />,
+        actionIconPlacement: 'right',
+        onActionClick: () => navigate(absoluteRoutes.studio.editChannel()),
```
Comment (rafalpawlow) [inline packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationSteps/YppAuthorizationRequirementsStep/YppAuthorizationRequirementsStep.tsx]: This is probably my code but I'm thinking if we shouldn't also switch the channel. Clicking on this link will take me to the current channel and not the one that doesn't meet the requirements.

**Review by rafalpawlow — COMMENTED** [review]
Something is wrong with authorization failed modal. Take a look

https://user-images.githubusercontent.com/13706246/193021779-47705ab5-4ca8-42c7-8c02-194d8e2fb0cc.mov

File: packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.hooks.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+  const handleGoogleAuthError = useCallback(
+    (error: string) => {
+      // Google auth failed, show error modal
+      SentryLogger.error('Google Auth failed', 'YppAuthorizationModal', error)
+
+      // TODO: use proper copy
+      openConfirmationModal({
+        title: 'Authorization failed',
+        description: <>Google Auth failed</>,
```
Comment (drillprop) [inline packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.hooks.tsx]: Can't we use regular string here?

File: packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.tsx [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+          description:
+            'Before you can apply to the program, make sure both your Atlas and YouTube channels meet the below conditions.',
+          primaryButton: {
+            text: 'Authorize with YouTube',
+            onClick: handleAuthorizeClick,
+            disabled: !isSelectedChannelValid,
+          },
+          component: (
+            <YppAuthorizationRequirementsStep
+              onChangeChannel={() => selectedChannel && setActiveUser({ channelId: selectedChannel?.id })}
```
Comment (drillprop) [inline packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.tsx]: We could either omit `?.` here or remove `selectedChannel &&`

File: packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.stories.tsx [hunk 5]
```diff
@@ -11,34 +14,29 @@ export default {
...
-      {
-        id: '2',
-        title: 'Joystream',
-        createdAt: new Date(),
-        follows: 987,
-        rewardAccount: '',
-        channelStateBloatBond: '',
-      },
-    ],
+    currentStepIdx: null,
```
Comment (drillprop) [inline packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.stories.tsx]: Can we add the `select` control here, or something more descriptive? 
image
Right now I'm not sure what I can type here. I mean, I can change the number, but in the end, I don't know what a particular number stands for.

**Review by drillprop — CHANGES_REQUESTED** [review]
Found some minor things. Besides that works good

File: packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.hooks.tsx (hunk 3 above)
Comment (kdembler) [reply packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.hooks.tsx]: Yeah, that's a leftover from the previous version of the copy

File: packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.stories.tsx (hunk 5 above)
Comment (kdembler) [reply packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.stories.tsx]: Fixed

File: packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.tsx (hunk 4 above)
Comment (kdembler) [reply packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.tsx]: Fixed

**Review by drillprop — APPROVED** [review]

**Pull request merged**